src/support/generation.c

```diff
@@ -182,13 +180,16 @@ __gen_drain_callback(
                       session, verbose_orig_level, WT_VERB_CHECKPOINT_PROGRESS, WT_VERBOSE_DEBUG_1);
                 }
                 cookie->verbose_timeout_flags = true;
-                /* Now we have enabled more logs, spin another time to get some information. */
+                /* Now we have enabled more logs, continue waiting to get more information. */
                 continue;
             }
-#endif
-            if (time_diff_ms >= conn->gen_drain_timeout_ms) {
+
+            if (!timeout_triggered && time_diff_ms >= conn->gen_drain_timeout_ms) {
```

Comment:
We need to log it continuously until we came out of that evict, so I don't think we need the `timeout_triggered` flag?

Reply:
Then it will log too many times. I'll think about it.